blog/_posts/2019-11-13-affiliations-feature.md

```diff
@@ -0,0 +1,67 @@
+		 	 	 		
+We were working with a dataset of approximately 34 million unique strings (about 4.5m astronomy and 29+ million physics). In the end, we were able to identify approximately 2.25 million affiliation strings, resulting in matching 93% of astronomy affiliations and 70% of physics affiliations to an institutional identifier.
+
+
+Automated Pipelines: assigning identifiers to specific author-affiliation-bibcode entities
+
+With the initial set of affiliation strings identified we, along with ADS backend programmer Stephen McDonald, designed an automated pipeline to add canonical affiliation identifiers to our database of 15 million references.  It’s a simple process.  First we normalize all of the strings with all uppercase letters, remove spaces and a subset of punctuation marks not needed for disambiguation, which results in a reduced number of matching strings; often, strings will be identical without punctuation, such as strings with “... U.S.A.” and “... USA”.  We then assign the strings to a dictionary, where the strings themselves are the keys, and the values are the identifiers.  If a normalized incoming string matches a key in the dictionary, we assign it the same aff_id, and the record with augmented affiliation data is sent back to our database of metadata.  The augmented affiliations are then passed to Solr and then to the ADS website with the next reindexing process.
+
+We’ve entirely automated the process, so once the dictionary of affiliation strings and matched identifiers is created, it’s a hands-off component of our metadata processing pipeline.  We update dictionaries about once a month, and occasionally more often if we have a large batch of incoming unmatched affiliation strings.  This process is similar to the first curation step, with the only difference being that we prioritize strings according to how many times they appear.  For example, if we receive a number of papers from collaborations with hundreds or thousands of authors *and* their affiliations don’t exactly match our dictionary, we need to add them.  Early on in this process, we came across strings that had many thousands (and sometimes tens of thousands) of occurrences, but the latest batch of unclassified strings appear fewer than 800 times in our entire database and this maximum frequency drops each time we create a new dictionary.
+
+However, we’re getting close to the limit of what we can do with human classification without lots of effort.  Our most recent pass through our metadata has just over 4.6 million unclassified strings.  There are about 6000 unclassified affiliations that appear 100 times or more in the metadata and assigning them aff_ids would provide a lot of new information.  However, over 95 percent of unclassified affiliations occur fewer than 10 times, and **46 percent of affiliation strings only appear once in our affiliation metadata**.
```

Comment:
Would one of the plots you made for the ADSUG work here? It'd be great to see some of this graphically.

Reply:
matched.jpg and unmatched.jpg